sources/BitMap.move

```diff
@@ -0,0 +1,103 @@
+module SFC::BitMap{
+    use StarcoinFramework::Vector;
+
+    struct Item has store, drop, copy {
+        index: u128,
+        bits: u128
+    }
+
+    struct BitMap has store, drop {
+        data: vector<Item>
+    }
+
+    public fun new(): BitMap {
+        BitMap {
+            data: Vector::empty<Item>()
+        }
+    }
+
+    public fun get(bitMap: &mut BitMap, index: u128): bool {
+        let itemIndex = index >> 7;
+        let mask = 1 << (index & 0x7f as u8);
+
+        let i = 0;
+        let v = &bitMap.data;
+        let len = Vector::length(v);
+        while (i < len) {
+            let item = Vector::borrow(v, i);
+            if (item.index == itemIndex) {
+                return item.bits & mask != 0
+            };
+            i = i + 1;
+        };
+        false
+    }
+
+    public fun set(bitMap: &mut BitMap, index: u128) {
+        let itemIndex = index >> 7;
+        let mask = 1 << (index & 0x7f as u8);
+
+        let i = 0;
+        let v = &mut bitMap.data;
+        let len = Vector::length(v);
+        while (i < len) {
+            let item = Vector::borrow_mut(v, i);
+            if (item.index == itemIndex) {
+                item.bits = item.bits | mask;
+                return
+            };
+            i = i + 1
+        };
+        Vector::push_back(&mut bitMap.data, Item { index: itemIndex, bits: mask })
+    }
+
+    public fun unset(bitMap: &mut BitMap, index: u128) {
+        let itemIndex = index >> 7;
```

Comment:
Why `itemIndex = index >> 7`, the bits is u128.

Reply:
I am using the upper 121 bit of key to discriminate items, and use a u128 `bits` to store the items